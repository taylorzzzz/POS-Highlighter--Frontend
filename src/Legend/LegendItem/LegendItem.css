.LegendItem {
  margin: 5px 0;
  transition: color .2s;
}

.LegendItem.editActive label {
  color: var(--color-text-disabled);
}
.LegendItem.unselected {
  color: var(--color-text-disabled);
}


.dot input[type="checkbox"] {
  position: absolute;
  opacity: 0;
}

.dot label {
  text-transform: capitalize;
  font-size: 14px;
  position: relative;
  cursor: pointer;
  padding: 0;
  user-select: none;
}

.dot label::before {
  content: '';
  margin-right: 10px;
  display: inline-block;
  vertical-align: middle;
  width: 10px;
  height: 10px;
  background: var(--color-disabled);
  border-radius: 100%;
}

.dot input[type="checkbox"]:hover + label:before {
  background: #ccc;
}

.dot input[type="checkbox"] {

  &:checked + label:before {
    background: #3344aa;
  }

  &#adjective:checked + label:before {
    background: var(--color-adjective);
  }

  &#adverb:checked + label:before {
    background: var(--color-adverb);
  }
  &#conjunction:checked + label:before {
    background: var(--color-conjunction);
  }
  &#determiner:checked + label:before {
    background: var(--color-determiner);
  }
  &#noun:checked + label:before {
    background: var(--color-noun);
  }
  &#number:checked + label:before {
    background: var(--color-number);
  }
  &#other:checked + label:before {
    background: var(--color-other);
  }
  &#preposition:checked + label:before {
    background: var(--color-preposition);
  }
  &#pronoun:checked + label:before {
    background: var(--color-pronoun);
  }
  &#punctuation:checked + label:before {
    background: var(--color-punctuation);
  }
  &#verb:checked + label:before {
    background: var(--color-verb);
  }

}


