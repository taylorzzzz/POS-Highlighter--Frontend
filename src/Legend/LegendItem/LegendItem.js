import React from 'react';

import './LegendItem.css';

function LegendItem({ pos, selected, toggleSelection }) {

  const unselected = selected ? '' : 'unselected';
  const classes = `${unselected} LegendItem`;

  return (
    <div className={classes}>

      <input 
        type="checkbox"  
        checked={ selected }
        onChange={ toggleSelection }
        data-pos={ pos }
        id={ pos }/>

      <label htmlFor={ pos }>{ pos }</label>

    </div>
  )

}

export default LegendItem;