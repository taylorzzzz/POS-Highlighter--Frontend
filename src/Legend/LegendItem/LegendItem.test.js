import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import LegendItem from './LegendItem';

test('LegendItem renders properly', () => {
    const tree = renderer.create(<LegendItem />).toJSON();
    
    expect(tree).toMatchSnapshot();
})

test('LegendItem toggleSelection events properly', () => {

    const toggleSpy = jest.fn();

    const wrapper = shallow(<LegendItem toggleSelection={toggleSpy}/>);
    
    expect(toggleSpy).toHaveBeenCalledTimes(0); 

    const input = wrapper.find('input');
    
    input.simulate('change');

    expect(toggleSpy).toHaveBeenCalled();
})