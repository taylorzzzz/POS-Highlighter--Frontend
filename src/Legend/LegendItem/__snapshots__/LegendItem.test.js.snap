// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendItem renders properly 1`] = `
<div
  className="unselected  LegendItem dot"
>
  <input
    checked={undefined}
    data-pos={undefined}
    id={undefined}
    onChange={undefined}
    type="checkbox"
  />
  <label
    htmlFor={undefined}
  />
</div>
`;
